public/app/percona/dbaas/components/DBCluster/EditDBClusterPage/DBClusterBasicOptions/Restore/Restore.tsx

```diff
@@ -0,0 +1,99 @@
+import { AsyncSelectField, validators } from '@percona/platform-core';
+import React, { FC, useState } from 'react';
+import { Field } from 'react-final-form';
+
+import { FieldSet, useStyles, Switch } from '@grafana/ui';
+
+import { useSelector } from '../../../../../../../types';
+import { SelectField } from '../../../../../../shared/components/Form/SelectField';
+import { getBackupLocations } from '../../../../../../shared/core/selectors';
+import { AddDBClusterFormValues } from '../../EditDBClusterPage.types';
+
+import { Messages } from './Restore.messages';
+import { RestoreService } from './Restore.service';
+import { getStyles } from './Restore.styles';
+import { RestoreFields, RestoreFromProps } from './Restore.types';
+
+export const Restore: FC<RestoreFromProps> = ({ form }) => {
+  const styles = useStyles(getStyles);
+
+  const [enableRestore, setEnableRestore] = useState(false);
+  const { result: locations = [], loading: locationsLoading } = useSelector(getBackupLocations);
+  const locationsOptions = locations.map((location) => ({
+    label: location.name,
+    value: location.locationID,
+  }));
+
+  const { restoreFrom, kubernetesCluster } = form.getState().values as AddDBClusterFormValues;
```

Comment:
This may show an TS error about type assertions (don't know if that's the case, but it might)

Reply:
Hmm... I'll try to add myTypes one else time there...becouse  I don't remember exactly what the problem that prevented me from doing this looked like

Reply:
I removed this prop, becouse of this error during covering form with types, i see that integrated alerting have the same problem...
![image](https://user-images.githubusercontent.com/90199600/216990964-78b0050a-0d05-4e13-b092-b9538598a693.png)

I think there is definitely a solution, but so far I can't find it. Can we potentially move this into a separate refactoring ticket if I'll not solve this issue during tonight?

Reply:
Sure.